src/detect-engine-alert.c

```diff
@@ -597,6 +598,12 @@ void PacketAlertFinalize(const DetectEngineCtx *de_ctx, DetectEngineThreadCtx *d
             p->flags |= PKT_FIRST_ALERTS;
         }
     }
+
+    /* Notify capture layer about packets with alerts, capture impl may
+     * update per-capture context (e.g. pcap-file alert counts). */
+    if (p->alerts.cnt > 0) {
```

Comment:
minor: I think it would be nice to merge this with the `PKT_FIRST_ALERTS` block right above it, so that we check for `p->alerts.cnt` only once